### DISCUSSION TITLE
socat error on Monday mornings only

### DISCUSSION THREAD

**Discussion by johnjohnk2** [gnzsnz/ib-gateway-docker#120] [Q&A] [answered]
#102 I am encountering a consistent issue on the container . Every Monday morning, the following error prompts a need for a manual container restart:

`2024/06/03 05:00:12 socat[63188] E connect(5, AF=2 127.0.0.1:4002, 16): Connection refused
.> 😘 Received SIGINT or SIGTERM. Shutting down IB Gateway.
.> Stopping Xvfb.
.> Stopping socat.
.> Stopping IBC.
IBC returned exit status 1
autorestart file not found
`

This error indicates that the connection to a local service on port 4002 is refused, triggering a shutdown sequence. Moreover, it appears the autorestart functionality is not working as expected, as evidenced by "autorestart file not found."

Could anyone please provide insights or suggest modifications to ensure the container can automatically recover from such errors without requiring manual intervention? This would significantly enhance stability and usability on ARM64 systems.

Thank you for any help or suggestions you can provide.

Best regards,

**Comment by gnzsnz** [comment]
are you restarting every sunday? if not is normal that on Monday it will not work.

**Comment by johnjohnk2** [comment]
Gonzalo, Why should I need to restart every Sunday? I just have the AUTO_RESTART_TIME=02:59 AM setting  in my .env

**Comment by gnzsnz** [comment, marked as answer]
you need to enter credentials once per week, on sunday will disconnect and require credentials. basically when you restart on mondays you are entering credentials again and works. if you do this on sunday, it will be ready on mondays.

**Comment by omair-rit** [comment]
Is there any way around entering credentials once per week?

**Comment by gnzsnz** [comment]
Entering credentials once per week is the way around entering credentials every day. So no, there is no way around it AFAIK.

**Comment by johnjohnk2** [comment]
But don't understand the difference between restaring the container and the AUTO_RESTART_TIME=02:59 AM parameter ? I thought the AUTO_RESTART_TIME was entering credentials to restart the gateway inside the container....

**Comment by gnzsnz** [comment]
some usefull links on this subject [IBC user guide](https://github.com/IbcAlpha/IBC/blob/master/userguide.md) and [3.19 release notes](https://github.com/IbcAlpha/IBC/releases/tag/3.19.0)

honestly IBC documentation is much clear than TWS's docs

**Comment by gnzsnz** [comment]
duplicate with #145
